src/import/load-costume.js

```diff
@@ -11,12 +11,8 @@ const loadVector_ = function (costume, costumeAsset, runtime, rotationCenter, op
         svgString = runtime.v2SvgAdapter.toString();
         // Put back into storage
         const storage = runtime.storage;
-        costumeAsset.encodeTextData(svgString, storage.DataFormat.SVG);
-        costume.assetId = storage.builtinHelper.cache(
-            storage.AssetType.ImageVector,
-            storage.DataFormat.SVG,
-            costumeAsset.data
-        );
+        costume.asset.encodeTextData(svgString, storage.DataFormat.SVG, true);
+        costume.assetId = costume.asset.assetId;
```

Comment:
I tried to make the changes for this move as little logic as possible and just update the API, to reduce variables so I didn't introduce regressions with the refactor. So in this case, the `encodeTextData` has the capability to set data, and generate a new assetId. But there may be cases where when it does this, it doesn't actually change the data that was already there. But that should have been true previously as well, but we were reassigning the assetId then too (the md5 was recalculated by the `cache` method instead).

I think after this is in and working, it would be a good idea to review the loading process and see if all of the information we're passing around is still necessary, or if we could just derive it from the asset when we needed to. I think we could also take that time to see if all the pre-existing logic is sound, or are we doing more work than we need to because we've done changes piecemeal and haven't looked at the process holistically in a while.

Reply:
That sounds like a really good idea. =]